Traceback (most recent call last):
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\jupyter_core\utils\__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\ibhan\miniconda3\envs\mfbook_310\lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
#run a solution using the revised model and replacing the dataframe with the new one (exostart)  (that includes the exogenized variables) with their values.  Store thre results in a dataframecalled Baseline
# they will also be stored in mpak.latestdf

update_variables = {'Carbon tax on Coal (USD/Ton)'       :'PAKGGREVCO2CER',
                    'Carbon tax on Oil (USD/Ton'         :'PAKGGREVCO2OER',
                    'Carbon tax on Natural Gas (USD/Ton' :'PAKGGREVCO2GER'}
value = 29

for tax,var in update_variables.items():
        alternative_exo.loc[2020:2100,var] = value  # sets the carbon tax on all three products to value (29)
     
        
res= mpak(alternative_exo,2020,2100,alfa = 0.7,silent=1,solver = 'sim',relconv = 0.00000001) #runs the solutiona nd saves it with the name to teh left
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mNameError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[15], line 10[0m
[0;32m      7[0m value [38;5;241m=[39m [38;5;241m29[39m
[0;32m      9[0m [38;5;28;01mfor[39;00m tax,var [38;5;129;01min[39;00m update_variables[38;5;241m.[39mitems():
[1;32m---> 10[0m         [43malternative_exo[49m[38;5;241m.[39mloc[[38;5;241m2020[39m:[38;5;241m2100[39m,var] [38;5;241m=[39m value  [38;5;66;03m# sets the carbon tax on all three products to value (29)[39;00m
[0;32m     13[0m res[38;5;241m=[39m mpak(alternative_exo,[38;5;241m2020[39m,[38;5;241m2100[39m,alfa [38;5;241m=[39m [38;5;241m0.7[39m,silent[38;5;241m=[39m[38;5;241m1[39m,solver [38;5;241m=[39m [38;5;124m'[39m[38;5;124msim[39m[38;5;124m'[39m,relconv [38;5;241m=[39m [38;5;241m0.00000001[39m) [38;5;66;03m#runs the solutiona nd saves it with the name to teh left[39;00m

[1;31mNameError[0m: name 'alternative_exo' is not defined

